replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/persistence/ReplicationPersistentStore.java

```diff
@@ -0,0 +1,153 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ditto.replication.api.impl.persistence;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Stream;
+import org.apache.commons.lang3.reflect.FieldUtils;
+import org.codice.ddf.persistence.PersistenceException;
+import org.codice.ddf.persistence.PersistentItem;
+import org.codice.ddf.persistence.PersistentStore;
+import org.codice.ditto.replication.api.ReplicationPersistenceException;
+import org.codice.ditto.replication.api.data.Persistable;
+import org.codice.ditto.replication.api.impl.data.AbstractPersistable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * ReplicationPersistentStore uses {@link Persistable}s to alter and retrieve data in a {@link
+ * PersistentStore}
+ */
+public class ReplicationPersistentStore {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ReplicationPersistentStore.class);
+
+  private static final int DEFAULT_PAGE_SIZE = 100;
+
+  private static final int DEFAULT_START_INDEX = 0;
+
+  private final PersistentStore persistentStore;
+
+  public ReplicationPersistentStore(PersistentStore persistentStore) {
+    this.persistentStore = persistentStore;
+  }
+
+  private <T extends AbstractPersistable> T createPersistable(Class<T> klass) {
```

Comment:
I think this class could just deal with `T extends Persistable` if you move the to/from map methods to persistable which seems cleaner to me.

Reply:
As mentioned above, we do not want to bleed the implementation into the API. Having the to/from on the impl side is appropriate.